frontend/src/metabase/public/components/EmbedModal/EmbedModalContent/EmbedModalContent.unit.spec.tsx

```diff
@@ -0,0 +1,189 @@
+
+    userEvent.click(
+      screen.getByRole("tab", {
+        name: "Parameters",
+      }),
+    );
+
+    expect(screen.getByText("My param")).toBeInTheDocument();
+    expect(screen.getByLabelText("My param")).toHaveTextContent("Disabled");
+  });
+
+  it("should render saved parameters", () => {
+    setup({
+      embedType: "application",
+      resource: {
+        ...createMockDashboard(),
+        embedding_params: {
+          my_param: "locked",
+        },
+      },
+      resourceParameters: [
+        {
+          id: "my_param",
+          name: "My param",
+          slug: "my_param",
+          type: "category",
+        },
+      ],
+    });
+
+    userEvent.click(
+      screen.getByRole("tab", {
+        name: "Parameters",
+      }),
+    );
+
+    const parametersSection = screen.getByLabelText(
+      "Enable or lock parameters",
+    );
+    expect(within(parametersSection).getByText("My param")).toBeInTheDocument();
+    expect(
+      within(parametersSection).getByLabelText("My param"),
+    ).toHaveTextContent("Locked");
+  });
+
+  it("should only render valid parameters", () => {
+    setup({
+      embedType: "application",
+      resource: {
+        ...createMockDashboard(),
+        embedding_params: {
+          old_param: "locked",
+        },
+      },
+      resourceParameters: [
+        {
+          id: "my_param",
+          name: "My param",
+          slug: "my_param",
+          type: "category",
+        },
+      ],
+    });
+
+    userEvent.click(
+      screen.getByRole("tab", {
+        name: "Parameters",
+      }),
+    );
+
+    expect(screen.getByText("My param")).toBeInTheDocument();
+    expect(screen.getByLabelText("My param")).toHaveTextContent("Disabled");
+  });
+
+  it("should update a card with only valid parameters", async () => {
+    const { mocks } = setup({
+      embedType: "application",
+      resource: {
+        ...createMockDashboard(),
+        embedding_params: {
+          old_param: "locked",
+        },
+      },
+      resourceParameters: [
+        {
+          id: "my_param",
+          name: "My param",
+          slug: "my_param",
+          type: "category",
+        },
+      ],
+    });
+
+    userEvent.click(
+      screen.getByRole("tab", {
+        name: "Parameters",
+      }),
+    );
+
+    expect(screen.getByText("My param")).toBeInTheDocument();
+    expect(screen.getByLabelText("My param")).toHaveTextContent("Disabled");
+
+    screen.getByLabelText("My param").click();
+
+    screen.getByText("Locked").click();
+
+    screen.getByRole("button", { name: "Publish changes" }).click();
+
+    await waitFor(() =>
+      expect(mocks.onUpdateEmbeddingParams).toHaveBeenCalledWith({
+        my_param: "locked",
+      }),
+    );
+  });
+});
+
+function setup({
+  embedType = null,
+  resource = {} as EmbedResource,
+  resourceType = "dashboard",
+  resourceParameters = [],
+  getPublicUrl,
+}: Partial<EmbedModalContentProps> = {}) {
+  const mocks = {
```

Comment:
Is this `mocks` objects necessary? I would prefer these to just be their own variables and then, in our tests, we can just destructure the mock functions that we need instead of dealing with the whole `mocks` object the whole time